app/state.go

```diff
@@ -1835,3 +1843,207 @@ func setBrokenArtifactFlag(store store.Store, artName string) {
 		// already in an error path.
 	}
 }
+
+type controlMapState struct {
+	wrappedState UpdateState
+	baseState
+}
+
+func NewControlMapState(wrapsState UpdateState) State {
+	return &controlMapState{
+		wrappedState: wrapsState,
+		baseState: baseState{
+			id: datastore.MenderStateUpdateControl,
+			t:  ToNone,
+		},
+	}
+}
+
+func (c *controlMapState) pauseName(t Transition) string {
+	switch t {
+	case ToArtifactReboot_Enter:
+		return "pause_before_rebooting"
+	case ToArtifactCommit_Enter:
+		return "pause_before_committing"
+	case ToArtifactInstall:
+		return "pause_before_installing"
+	}
+	panic(fmt.Sprintf("No pause name mapping for: %s. This is a logic error in the state machine code", t))
+}
+
+func (c *controlMapState) mapStateToName(t Transition) string {
+	switch t {
+	case ToArtifactReboot_Enter, ToArtifactCommit_Enter:
+		return c.wrappedState.Transition().String()
+	case ToArtifactInstall:
+		return "ArtifactInstall_Enter"
+	}
+	panic(fmt.Sprintf("No state name mapping for: %s. This is a logic error in the state machine code", t))
```

Comment:
I think this is fair, as it should catch logic errors during testing, but feel free to object.